packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -161,6 +164,22 @@ export const render = async (req, res, next) => {
     // Step 2 - Get the component
     const component = await route.component.getComponent()
 
+    // Step 2.5 - Prepass
+    console.log('Prepass Enabled? ', !!config.ssrParameters.enablePrepass)
+    if (!!config.ssrParameters.enablePrepass) {
```

Comment:
I believe we're enabling prepass as an opt-in feature to avoid creating a breaking change.

`react-ssr-prepass` seems to be in [experimental maintenance status](https://github.com/FormidableLabs/react-ssr-prepass#maintenance-status). 

There's a note in the Readme that says:
>  This library purely exists since react-dom/server does not support data fetching or suspense yet. This two-pass approach should just be used until server-side suspense support lands in React.
https://github.com/FormidableLabs/react-ssr-prepass#maintenance-status:~:text=This%20two%2Dpass%20approach%20should%20just%20be%20used%20until%20server%2Dside%20suspense%20support%20lands%20in%20React.

React 18 now supports suspense:
> In this release, we’re revamping our react-dom/server APIs to fully support Suspense on the server and Streaming SSR
https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#:~:text=we%E2%80%99re%20revamping%20our%20react%2Ddom/server%20APIs%20to%20fully%20support%20Suspense%20on%20the%20server%20and%20Streaming%20SSR

That means that for users using React 18, the purpose of `react-ssr-prepass` is not that relevant anymore.


Upgrading to React 18 will be a major version for us, but it can save us from implementing an interim solution using `react-ssr-prepass`.

The main motivation to use allow async calls server-side using hooks is the work of extracting the hooks library from the template to help partners to upgrade their projects.

Should we consider the Spike React 18 along this option to compare the efforts even though we know that upgrading to React 18 will be a major version bump?

Reply:
We're not quite using `react-ssr-prepass` for it's intended purpose since we're not using Suspense. But outside of it suspending whenever it encounters a thrown promise, it is just a library that provides a mechanism for doing two pass renders. 

I believe we will continue to use a two pass render approach even after we upgrade to React 18 so we can likely keep this separate from our upcoming React 18 spike.

Reply:
@vcua-mobify Suppose we decide to use React 18, we'll likely use the native React API to resolve data fetching while rendering. This will deprecate and replace the requirement for two render pass functionality provided by `react-ssr-prepass`.